fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/synthetic/SyntheticRecordPlannerTest.java

```diff
@@ -619,6 +639,40 @@ public void joinIndex() throws Exception {
             List<Tuple> results3 = recordStore.scanIndex(index, IndexScanType.BY_VALUE, range, null, ScanProperties.FORWARD_SCAN).map(IndexEntry::getKey).asList().join();
             assertEquals(expected3, results3);
         }
+
+
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore recordStore = recordStoreBuilder.setContext(context).open();
+
+            final QueryPlanner planner = setupPlanner(recordStore, null);
+
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType("Simple_Other")
+                    .setFilter(Query.field("simple").matches(Query.field("str_value").equalsValue("even")))
+                    //.setRequiredResults(ImmutableList.of(field("simple").nest("str_value"), field("other").nest("num_value_3")))
+                    .build();
+
+            // Scan([[photos],[photos]])
+            RecordQueryPlan plan = planner.plan(query);
```

Comment:
Is that the plan that gets chosen by the planner?

Reply:
yeah